### ISSUE TITLE
types / source information

### ISSUE THREAD

**Issue by rodhash** [linux-cultist/venv-selector.nvim#139]
Hi

New version looks very very nice, thanks and kudos

This isn't an issue per se, but I noticed that the type (or source) shows up only when I have my terminal in full screen .. this isn't visible when I have my temrinal in normal size or snapped to the left / right of my monitor..

Is this expected? I find this piece of info very nice to have but most of the time I'm not using the terminal in full screen.

<img width="1003" alt="image" src="https://github.com/linux-cultist/venv-selector.nvim/assets/29671981/8dd29bfe-64f4-4394-989f-ea55745aff6f">

**Comment by linux-cultist** [comment]
I agree, it would be nice if the UI could take into account screen size and adapt to it. I would like it to be visible even when the terminal is snapped to a side, at least on most normal resolutions.

Let me see if I can make the UI a bit smarter, it's a good idea. :)

**Comment by linux-cultist** [comment]
Update the plugin now and see if it looks better?

**Comment by tranbinh1604** [comment]
![CleanShot 2024-06-07 at 22 10 18](https://github.com/linux-cultist/venv-selector.nvim/assets/162952043/0a458959-3ada-4cb3-a5f9-fadf26441618)

mine after updated, I can not see the right project folder at all, also don't know which one to activate

**Comment by linux-cultist** [comment]
Its because you have so long paths and the window is smaller so it doesnt really fit. 

You can replace the entire first part of your results with something much shorter by adding a callback function that gets called by VenvSelect for each result it finds. Here is an example you can put in your config to play with.

```
  {
    "linux-cultist/venv-selector.nvim",
    dependencies = {
      "neovim/nvim-lspconfig",
      { "nvim-telescope/telescope.nvim", branch = "0.1.x", dependencies = { "nvim-lua/plenary.nvim" } },
      "mfussenegger/nvim-dap",
      "mfussenegger/nvim-dap-python",
    },
    lazy = false,
    dev = true,
    branch = "regexp",
    config = function()
   
      local function shorter_name(filename)
        -- Define replacements as key-value pairs
        local replacements = {
          ["/Users/binhtt/Library/Mobile Documents/com~apple~CloudDocs/Coding"] = ".../Coding",
        }

        -- Loop through the table and replace each found path with its specific replacement
        for path, replacement in pairs(replacements) do
          filename = string.gsub(filename, path, replacement)
        end

        return filename
      end

      require("venv-selector").setup {
        settings = {
          options = {
            on_telescope_result_callback = shorter_name,
          },
        },
      }
    end,
    keys = {
      { ",v", "<cmd>VenvSelect<cr>" },
    },
  },

```

**Comment by linux-cultist** [comment]
I should probably add some built-in way to limit the result length as well. Normally its much more useful to see the end of the path rather than the beginning of the path, if there is not enough space to show it all.

**Comment by rodhash** [comment]
Thanks for the prompt feedback .. 2 things:

1. I've updated my plugin to the latest version but I see no change regarding the placement of the type:

<img width="277" alt="image" src="https://github.com/linux-cultist/venv-selector.nvim/assets/29671981/99060d82-4874-43ee-a396-babd266b2d63">


For me it still showing up only when full screen.


2. I liked the callback idea but it seems to not be working for me however I managed to shorten my path using a little bit of **sed**:
```lua
          search = {
            venvs = {
              command = "fd python$ ~/venv -I -L -E /proc | sed 's/.*venv/foobar/g'",
              type = "pyenv"
            },
...
```
<img width="664" alt="image" src="https://github.com/linux-cultist/venv-selector.nvim/assets/29671981/543e2a65-ee94-49e9-b428-88074000e02a">

With paths shortened I still can't see the type .. it shows up only with terminal in full screen

**Comment by rodhash** [comment]
Not sure if this is something that could be interesting to many, but, with an even uglier **sed** we could show up only the virtual env name and nothing else:

<img width="1572" alt="image" src="https://github.com/linux-cultist/venv-selector.nvim/assets/29671981/673fcf95-40a6-4803-98e0-deabcd4ca926">

Here how I did it:
```lua
miniconda = {
  command = "fd python$ /opt/homebrew/Caskroom/miniconda/base/envs -I -L -E /proc| sed -e 's_.*envs/\\\\(.*\\\\)/bin.*_\\\\1_g' -e 's_.*include.*__g'",
  type = "miniconda"
},
```

ps: not really sure why it's required to escape this much but it worked .. maybe later I'll try to understand what happend there usually just a single `\` would be enough

**Comment by linux-cultist** [comment]
I even tested the code above, it worked fine for me (but different path or course). It's possible that the tilde characters in the paths are not liked by lua though. If I had to guess, it's something like that. Using sed works but it will be slower.

The actual column where the type shows up is set to a specific width so I guess it's outside of the terminal width if it doesnt show up. Maybe a good idea to make this configurable in a setting since it doesn't seem like the telescope viewer can adapt the column width dynamically.

If you had a setting where you could set a number where the column would start, you could find a setting that works for you. :)

**Comment by rodhash** [comment]
sure.. thanks again!

btw.. while playing around I got this:

<img width="944" alt="image" src="https://github.com/linux-cultist/venv-selector.nvim/assets/29671981/1fa09fd4-2f13-4d52-bcb8-ceac632ed705">


I was not aware of the performance impact .. also not sure if this is going to be interesting to many but if we make the **sed** a little bit uglier we can show up only the venv name
